admittance_controller/src/admittance_controller.cpp

```diff
@@ -621,15 +613,16 @@ void AdmittanceController::read_state_from_hardware(trajectory_msgs::msg::JointT
 }
 
 bool AdmittanceController::read_state_from_command_interfaces(
-  trajectory_msgs::msg::JointTrajectoryPoint & state)
+  trajectory_msgs::msg::JointTrajectoryPoint & output_state)
 {
```

Comment:
Please leave a comment in the header file about what this function does. As far as I can tell, it does this:

// If velocities and positions are both available from the joint command interface, set output_state equal to them.
// If velocities or positions are unknown, output_state is unchanged and the function returns false.

Reply:
A small optimization might be possible:

If `has_values` is false at L644, we could return immediately, right?

Reply:
What if positions do not "have values" but velocities does "have values"?

With the present logic, I think the output would be wrong:

- empty state.positions, `has_values == false`
- state.velocities is filled
- `output_state = state;` never gets set, so `output_state` does not change
- the function returns false

Reply:
The method is not generic. It will be probably moved from here anyway in the future. So currently only supporting needed use-case.

Reply:
OK. Let's add a big TODO since we know this could be broken for some cases, then.

Reply:
Comments are added.